go/src/koding/klient/machine/client/supervised_test.go

```diff
@@ -0,0 +1,67 @@
+package client_test
+
+import (
+	"testing"
+	"time"
+
+	"koding/klient/machine/client"
+	"koding/klient/machine/client/testutil"
+)
+
+func TestSupervisedWait(t *testing.T) {
+	serv := &testutil.Server{}
+
+	dc, err := client.NewDynamic(testutil.DynamicOpts(serv, testutil.NewBuilder(nil)))
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer dc.Close()
+
+	dcf := func() (client.Client, error) { return dc.Client(), nil }
+
+	// Server is off, so the timeout should be reached.
+	const timeout = 50 * time.Millisecond
+	spv := client.NewSupervised(dcf, timeout)
+	startC, hitC := make(chan struct{}), make(chan time.Time)
+
+	go func() {
+		<-startC
+		_, err = spv.CurrentUser()
+		hitC <- time.Now()
+	}()
+
+	now := time.Now()
+	close(startC)
+	select {
+	case called := <-hitC:
+		if tret := called.Sub(now); tret < timeout {
+			t.Fatalf("want return at least after %v; got %v", timeout, tret)
+		}
+		if err != client.ErrDisconnected {
+			t.Fatalf("want err = %v; got %v", client.ErrDisconnected, err)
+		}
+	case <-time.After(time.Second):
+		t.Fatalf("test timed out after 1s")
+	}
+
+	// Server starts to be responsive while Supervisor waits for it.
+	spv = client.NewSupervised(dcf, 2*time.Second)
+	startC, hitC = make(chan struct{}), make(chan time.Time)
+
+	go func() {
+		<-startC
+		_, err = spv.CurrentUser()
+		hitC <- time.Now()
+	}()
+
+	close(startC)
+	serv.TurnOn()
+	select {
+	case <-hitC:
+		if err == client.ErrDisconnected {
+			t.Fatalf("want err != %[1]v; got %[1]v", client.ErrDisconnected)
```

Comment:
"want err != client.ErrDisconnected" is enough msg, if it fails it's obvious why

(the real reason is that I always need to lookup the doc whether positional arguments are 0- or 1-index thus I try to avoid it if possible)